sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/json/JsonFileFormat.scala

```diff
@@ -102,6 +102,15 @@ class JsonFileFormat extends TextBasedFileFormat with DataSourceRegister {
       sparkSession.sessionState.conf.sessionLocalTimeZone,
       sparkSession.sessionState.conf.columnNameOfCorruptRecord)
 
+    // Check a field requirement for corrupt records here to throw an exception in a driver side
+    dataSchema.getFieldIndex(parsedOptions.columnNameOfCorruptRecord).map { corruptFieldIndex =>
```

Comment:
And.. not a strong preference but maybe put this in `JacksonUtils` and rename it `JsonUtils` if @cloud-fan is okay? 

Maybe we could throws an exception as `IlligalArgumentException` in the first place and then capture the message with `AnalysisException` (as `JacksonUtils.verifySchema` is doing in `StructToJson`). This is not a strong opinion too.